accounts-db/src/accounts_index/in_mem_accounts_index.rs

```diff
@@ -1008,24 +1013,47 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> InMemAccountsIndex<T,
         // this could be tunable dynamically based on memory pressure
         // we could look at more ages or we could throw out more items we are choosing to keep in the cache
         if Self::should_evict_based_on_age(current_age, entry, startup, ages_flushing_now) {
-            if entry.ref_count() != 1 {
-                Self::update_stat(&self.stats().held_in_mem.ref_count, 1);
-                (false, None)
-            } else {
-                // only read the slot list if we are planning to throw the item out
-                let slot_list = entry.slot_list.read().unwrap();
-                if slot_list.len() != 1 {
-                    if update_stats {
-                        Self::update_stat(&self.stats().held_in_mem.slot_list_len, 1);
+            match entry.ref_count() {
+                1 | 2 => {
+                    // only read the slot list if we are planning to throw the item out
+                    let slot_list = entry.slot_list.read().unwrap();
+                    let mut cached = false;
+                    let mut oldest_slot = Slot::MAX;
+                    let mut newest_slot = Slot::MIN;
+                    for (slot, info) in slot_list.iter() {
+                        cached = cached || info.is_cached();
+                        oldest_slot = oldest_slot.min(*slot);
```

Comment:
i don't think we care if oldest or newest is older than an epoch. Also, we can't write a slotlist.len() = 2 easily at all. I think the refcount==2, slot_list.len() == 1 is writable with 1 bit.

Reply:
Age is only u8. Even with max age to flush, there is going to be 102s max (255*400ms). I was thinking this is too frequent. 
After we flush these ref_count > 1 to disk, someone may touch them again, and we bring it to memory.
That's why I added one epoch old check, to make sure they are really cold (i.e. in ancient storage) then we flush.

Reply:
I'm confused with the connection to age here. Age is the u8 used to try to write items to disk in a bucket.

Reply:
My understanding as follows. 

Every 400ms, age increase by 1. 
For each entry inserted, it will set it age_to_flush based on whether it is in cache
  if not in cache, current_age + 5 
  if in cache, current_age + 255

So ref_count >2 entry if it is not in cache, after 5 age, then it is about 2s. then it will be flushed to disk.

Are these correct?